### DISCUSSION TITLE
Where to get frame from webcam ? Hope this is last question :D

### DISCUSSION THREAD

**Discussion by Samuel-Bachorik** [ultralytics/yolov5#2457] [General]
Hi, I'm currently working with yolo and webcam, so source is equal to "0". The problem is I need to access the webcam frames that is your Yolo going to process. I'm running detect.py. My goal is to get that frame process it for road detection and then I want to sent this frame to Yolo so I am asking you if you can show me the line of code where I can get webcam frames before they go to Yolo so I can process them first. 

THANK YOU FOR ALL OF YOUR HELP :)

**Comment by Samuel-Bachorik** [comment]
UPDATE

Best thing would be to process frame that your program is reading at abaolute begining but I dont know where it is . 

Or maybe better question is where can I find line of code that is reading frames from camera ?

**Comment by glenn-jocher** [comment]
@Samuel-Bachorik I don't understand the question. For webcam inference you can simply run
`python detect.py --source 0`

**Reply by Samuel-Bachorik** [reply]
@glenn-jocher Hello thank you for reply. I will try to explain this again. In your source code (somewhere) of Yolo i want to find where you are taking frames from webcam. For example, imagine i want to add watermark to yolo webcam output. I need to process every frame from webcam and add that watermark to every frame and i want to do something similar but i dont know where is that frame i can process.

**Reply by glenn-jocher** [reply]
@Samuel-Bachorik detect.py dataloader is set here:
https://github.com/ultralytics/yolov5/blob/747c2653eecfb870b1ed40b1e00e0ef209b036e9/detect.py#L45-L54

and dataloader outputs are here:
https://github.com/ultralytics/yolov5/blob/747c2653eecfb870b1ed40b1e00e0ef209b036e9/detect.py#L63-L69

**Reply by Samuel-Bachorik** [reply]
@glenn-jocher  Thank you, when I try to process img variable from line 63 then yolo is not working. Not working i mean it is running but not showing boxes. Only my changed frames. 

this is what i do
`for path, img, im0s, vid_cap in dataset: 
`
`           img = MyFunction(img)`
This "MyFunction just add road to this image and returns back image with same shape as img have in for loop (480, 640, 3). 
When i process this "img" then it is not working. When i run detect.py it is running and showing output but not any boxes only mine processed images. 

I absolutely dont understand why it is not working because all what i do is just change some pixels on this "img" nothing else 

**img is frame from webcam and all i do is i change few pixels on it and return back. On Yolo output i see my pixels changed but no yolo detection**

**Reply by glenn-jocher** [reply]
@Samuel-Bachorik the scope of support here is mainly for bug fixes and feature suggestions. I think this custom functionality naturally falls on you to implement, good luck.
